### PR TITLE
pkg/api/kubegraph/analysis: fix TestOverlappingHPAsWarning

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by sttts** [comment]
@DirectXMan12 ptal.
@mfojtik second commit is a cherry-pick candidate for master.

File: pkg/api/kubegraph/edges.go [hunk 1]
```diff
@@ -237,7 +237,7 @@ func AddHPAScaleRefEdges(g osgraph.Graph) {
 		switch groupVersionResource.GroupResource() {
 		case kapi.Resource("replicationcontrollers"):
 			syntheticNode = kubegraph.FindOrCreateSyntheticReplicationControllerNode(g, &kapi.ReplicationController{ObjectMeta: syntheticMeta})
-		case deployapi.Resource("deploymentconfigs"):
+		case deployapi.Resource("deploymentconfigs"), deployapi.LegacyResource("deploymentconfigs"):
```
Comment (mfojtik) [inline pkg/api/kubegraph/edges.go:240]: @sttts how about switching to https://github.com/openshift/origin/blob/master/pkg/deploy/api/register.go#L55

Comment (sttts) [reply pkg/api/kubegraph/edges.go:240]: Didn't know this exists ;)

**Review by DirectXMan12 — APPROVED** [review]
LGTM :+1:

**Comment by DirectXMan12** [comment]
@sttts I'll cherry-pick the second commit over to master once this merges, as per our IRC discussion.

File: pkg/api/kubegraph/edges.go (hunk 1 above)
Comment (sttts) [reply pkg/api/kubegraph/edges.go:240]: Was this added late to the branch? Cannot find it yet in the version in the rebase.

**Comment by sttts** [comment]
@DirectXMan12 merged here. Feel free to do the `IsResourceOrLegacy` switch in master. Will rebase onto that later.

**Pull request merged**